app/api/v1/resources_pob.rb

```diff
@@ -0,0 +1,70 @@
+module API
+  module V1
+    # Default options for all API endpoints and versions
+    module ResourcesPob
+      extend ActiveSupport::Concern
+
+      included do
+        desc 'Takes a resource and an organization id, signs the resource '\
+             'on behalf of an organization, and publishes a new envelope with '\
+             'that signed resource',
+             http_codes: [
+               { code: 201, message: 'Envelope created' },
+               { code: 200, message: 'Envelope updated' }
+             ]
+
+        params do
+          use :update_if_exists
+          use :skip_validation
+        end
+
+        # rubocop:disable Metrics/BlockLength
+        post 'resources/organizations/:organization_id/documents' do
+          authenticate!
+          params[:envelope_community] = select_community
+
+          organization = Organization.find(params[:organization_id])
+          publisher = current_user.publisher
+          organization_publisher = OrganizationPublisher
+                                   .where(organization: organization)
+                                   .where(publisher: publisher)
+                                   .first
+
+          unless organization_publisher
+            error!(
+              'User\'s publisher is not authorized to publish on behalf of this organization',
+              422
+            )
+          end
+
+          key_pair = organization_publisher.key_pairs.first
+
+          envelope_attributes = {
+            'envelope_type': 'resource_data',
+            'envelope_version': '1.0.0',
+            'envelope_community': 'ce_registry',
+            'resource': JWT.encode(
+              env['api.request.body'],
+              OpenSSL::PKey::RSA.new(key_pair.private_key),
+              'RS256'
+            ),
+            'resource_format': 'json',
+            'resource_encoding': 'jwt',
+            'resource_public_key': key_pair.public_key,
+            'organization_id': organization.id,
+            'publisher_id': publisher.id
+          }
+
+          envelope, errors = EnvelopeBuilder.new(envelope_attributes).build
```

Comment:
It's missing the `update_if_exists` and `skip_validation` params to the envelope_builder (if we want those feature, which is a good idea). See the envelopes api post method.